x-pack/plugins/infra/public/containers/logs/log_source/log_source.ts

```diff
@@ -29,47 +34,78 @@ export {
   LogSourceStatus,
 };
 
-export const useLogSource = ({ sourceId, fetch }: { sourceId: string; fetch: HttpHandler }) => {
+export const useLogSource = ({
+  sourceId,
+  fetch,
+  indexPatternsService,
+}: {
+  sourceId: string;
+  fetch: HttpHandler;
+  indexPatternsService: IndexPatternsContract;
+}) => {
   const getIsMounted = useMountedState();
   const [sourceConfiguration, setSourceConfiguration] = useState<
     LogSourceConfiguration | undefined
   >(undefined);
 
+  const [resolvedSourceConfiguration, setResolvedSourceConfiguration] = useState<
+    ResolvedLogSourceConfiguration | undefined
+  >(undefined);
+
   const [sourceStatus, setSourceStatus] = useState<LogSourceStatus | undefined>(undefined);
 
   const [loadSourceConfigurationRequest, loadSourceConfiguration] = useTrackedPromise(
     {
       cancelPreviousOn: 'resolution',
       createPromise: async () => {
-        return await callFetchLogSourceConfigurationAPI(sourceId, fetch);
+        const { data: sourceConfigurationResponse } = await callFetchLogSourceConfigurationAPI(
+          sourceId,
+          fetch
+        );
+        const resolvedSourceConfigurationResponse = await resolveLogSourceConfiguration(
```

Comment:
Yep, I added a note in the testing section:

> NOTE: The scenario of a referenced Kibana Index Pattern being deleted isn't handled in this PR (I don't think that was the expectation, but just to make it explicit. It will make more sense to pull that together alongside the UI branch).

I think it makes sense to address this on your branch (or a further branch afterwards). We discussed this briefly on a call, but I think it makes sense to handle this as we do the "no indices" checks. 

Users won't actually be able to set a KIP on their sourcce configuration with the work here, so I feel okay with the way it acts.